server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -155,7 +158,8 @@ public Response getDatabaseSegmentDataSource(@PathParam("dataSourceName") final
   @Produces(MediaType.APPLICATION_JSON)
   public Response getDatabaseSegments(
```

Comment:
I think this method is too big now, it should be split into smaller methods.

Reply:
Not addressed

Reply:
I looked at this, not sure best way to split it, took out parts of finding and authorizing `SegmentWithOvershadowedStatus` into a helper method.